dsl_parser/tests/test_get_attributes_list.py

```diff
@@ -0,0 +1,262 @@
+        result = functions.evaluate_functions(payload, {}, storage)
+
+        attributes_list = result['a']
+        assert attributes_list == ['gonna']
+
+    def test_get_attributes_list_no_values(self):
+        node_instances = [
+            {
+                'id': 'badnode_0hn011',
+                'node_id': 'badnode',
+                'runtime_properties': {'sing': 'forget'},
+            },
+            {
+                'id': 'goodnode_45713y',
+                'node_id': 'goodnode',
+            },
+        ]
+        nodes = [
+            {'id': 'badnode'},
+            {'id': 'goodnode'},
+        ]
+        storage = self._mock_evaluation_storage(node_instances, nodes)
+        payload = {'a': {'get_attributes_list': ['goodnode', 'sing']}}
+        result = functions.evaluate_functions(payload, {}, storage)
+
+        attributes_list = result['a']
+        assert attributes_list == []
+
+    def test_get_attributes_list_one_value(self):
+        node_instances = [
+            {
+                'id': 'badnode_0hn011',
+                'node_id': 'badnode',
+                'runtime_properties': {'sing': 'forget'},
+            },
+            {
+                'id': 'goodnode_45713y',
+                'node_id': 'goodnode',
+                'runtime_properties': {'sing': 'gonna'},
+            },
+        ]
+        nodes = [
+            {'id': 'badnode'},
+            {'id': 'goodnode'},
+        ]
+        storage = self._mock_evaluation_storage(node_instances, nodes)
+        payload = {'a': {'get_attributes_list': ['goodnode', 'sing']}}
+        result = functions.evaluate_functions(payload, {}, storage)
+
+        attributes_list = result['a']
+        assert attributes_list == ['gonna']
+
+    def test_get_attributes_list_no_node_instances(self):
+        node_instances = [
+            {
+                'id': 'badnode_0hn011',
+                'node_id': 'badnode',
+                'runtime_properties': {'sing': 'forget'},
+            },
+        ]
+        nodes = [
+            {'id': 'badnode'},
+            {'id': 'goodnode'},
+        ]
+        storage = self._mock_evaluation_storage(node_instances, nodes)
+        payload = {'a': {'get_attributes_list': ['goodnode', 'sing']}}
+        result = functions.evaluate_functions(payload, {}, storage)
+
+        attributes_list = result['a']
+        assert attributes_list == []
+
+    def test_get_attributes_list_no_node(self):
+        node_instances = [
+            {
+                'id': 'badnode_0hn011',
+                'node_id': 'badnode',
+                'runtime_properties': {'sing': 'forget'},
+            },
+        ]
+        nodes = [
+            {'id': 'badnode'},
+        ]
+        storage = self._mock_evaluation_storage(node_instances, nodes)
+        payload = {'a': {'get_attributes_list': ['goodnode', 'sing']}}
+        with testtools.testcase.ExpectedException(
+                KeyError,
+                '.*Node not found.*'):
```

Comment:
I'd expect the `.*` at the beginning and end are implied, at least that's how other raisesRegexp methods work?

Reply:
It's not in this one, as I discovered earlier today.